### PR TITLE
Create RunningCompactionInfo in core for reuse

### PR DESCRIPTION
**Pull request by milleruntime**
* Create RunningCompactionInfo in core to be reused outside the Monitor
* Moves the logic for parsing progress from running compactions to core so that
server utilities can get the progress as well

### CODE REVIEW

**Comment by DomGarguilo** [comment]
> * Moves the logic for parsing progress from running compactions to core so that
>       server utilities can get the progress as well

Were there any current server utilities that would benefit from this information or are these changes mostly to make the information more accessible for future changes?

File: server/monitor/src/main/java/org/apache/accumulo/monitor/rest/compactions/external/CompactorInfo.java [hunk 1]
```diff
@@ -21,11 +21,9 @@
...
 
   // Variable names become JSON keys
-  public long lastContact = 0L;
-  public String server = "";
-  public String queueName = "";
-
-  public CompactorInfo() {}
+  public long lastContact;
+  public String server;
+  public String queueName;
```
Comment (ctubbsii) [inline server/monitor/src/main/java/org/apache/accumulo/monitor/rest/compactions/external/CompactorInfo.java:26]: Should these public variables be final?

**Comment by milleruntime** [comment]
> > * Moves the logic for parsing progress from running compactions to core so that
> >   server utilities can get the progress as well
> 
> Were there any current server utilities that would benefit from this information or are these changes mostly to make the information more accessible for future changes?

The `listcompactions` shell command is the only one that comes to mind. But I was planning on creating a new `ec-admin` command to hold all the commands for external compactions, to include listing running compactions.

File: server/monitor/src/main/java/org/apache/accumulo/monitor/rest/compactions/external/CompactorInfo.java (hunk 1 above)
Comment (milleruntime) [reply server/monitor/src/main/java/org/apache/accumulo/monitor/rest/compactions/external/CompactorInfo.java]: Fixed in 149e858

**Comment by milleruntime** [comment]
I am going to merge this so I can continue working on related changes. Let me know if there was anything else you saw.

**Pull request merged**